### ISSUE TITLE
Call operator `&` does not ignore argument variable with an empty string anymore in v7.3.0

### ISSUE THREAD

**Issue by Simran-B** [PowerShell/PowerShell#18761]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://docs.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

I use a PowerShell script to call another script. It optionally forwards arguments using a variable initialized to an empty string. With pwsh 7.2.7, this works fine. The call operator does not forward the empty string.

With 7.3.0, the behavior of the call operator is different. It does forward the empty string. I'm not sure if this change is intentional - if it is, is this documented somewhere? If not, consider moving this issue to the docs repo.

`caller.ps1`:

```ps
$mandatory = 'hello'
$optional = ''
& 'pwsh' 'callee.ps1' $mandatory $optional
```

`callee.ps1`:

```ps
foreach ($arg in $args) {
    Write-Host "'$arg' ($($arg.Length))"
}
```

The workaround/fix I found for 7.3.0 is to initialize `$optional` to `$null` instead.

### Expected behavior

```console
PS> .\caller.ps1
'hello' (5)
```

### Actual behavior

```console
PS> .\caller.ps1
'hello' (5)
'' (0)
```

### Environment data

```powershell
Name                           Value
----                           -----
PSVersion                      7.3.0
PSEdition                      Core
GitCommitId                    7.3.0
OS                             Microsoft Windows 10.0.22621
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Comment by 237dmitry** [comment]
If we take this specific json-example, then in Windows, then the parameter is passed correctly in `Legacy` and `Windows` modes. In Linux, the parameter is passed correctly in `Windows` and `Standard` modes.

It turns out that `Legacy` and `Standard` behave differently on different platforms. They were mixed up?

**Comment by jborean93** [comment]
Linux doesn't suffer from the same problem as on Windows. With Windows PowerShell needs to translate an array of arguments into a single string for the command line. It originally did this incorrectly with Legacy mode where essentially the logic was:

* Join the array with a string
* Wrap each entry with double quotes if it contained whitespace

Now on non-Windows, for PowerShell to start a process all it needs to do is to provide the array of arguments as a string so no escaping is needed. As it doesn't need to convert that array to a single string the json string itself is given directly to the process as is without any escaping needed to break the json value.

**Comment by mklement0** [comment]
@237dmitry 

Your table conflates the _raw process command line_ - a Windows-only concept - with _arguments received verbatim_ - a Unix-only concept.

> in Windows, then the parameter is passed correctly in `Legacy` and `Windows`

No, it isn't. `Legacy` is just as broken on Window as it is on Unix (note that on Unix `Windows` is in effect the same as `Standard`).

Note that passing a JSON string to `cmd /c echo` - e..g `cmd /c echo '{ "abc": "cba" }'` - is _not_ a valid test, because it involves virtually no parsing by `cmd`.

In fact, it shows you how `Legacy` mode is _broken_, by showing you part of the _raw_ command line:

`"{ "abc": "cba" }"` was (brokenly) passed, exactly as described by @jborean93: because the argument contained _spaces_, it was _blindly_ wrapped in `"..."`, without proper escaping of the _embedded_ `"` chars.

CLIs that follow the Microsoft C/C++ / `CommandLineToArg()` command-line parsing rules parse `"{ "abc": "cba" }"` into an argument as follows:

  * Verbatim `{ `, concatenated with verbatim `abc`, concatenated with verbatim `cba`, concatenate with verbatim ` }`, yielding verbatim `{ abc: cba }`, i.e. effective loss of all `"` chars.

  * In other words: _All_ `"` characters were considered to have _syntactic_ function and were therefore _removed_; the _lack of escaping_ of what were meant to be _embedded_  `"` chars. meant that they were removed too.

  * If you don't have an executable handy that echoes the _parsed_ arguments based on those rules, you can use `choice.exe` in a pinch, which will show the effective loss of all `"` chars (ignore the ` [Y,N]?Y` suffix in the output):

        choice /d Y /t 0 /m '{ "abc": "cba" }'  # -> { abc: cba } [Y,N]?Y

**Comment by 237dmitry** [comment]
> Note that passing a JSON string to cmd /c echo - e..g cmd /c echo '{ "abc": "cba" }' - is not a valid test, because it involves virtually no parsing by cmd.

![Screenshot 2022-12-12 144412](https://user-images.githubusercontent.com/78153320/207037366-e708ea35-5012-45d1-9f15-3a0d59149ece.png)

 > Legacy is just as broken on Window

Then that explains it all. Without this discussion, I would never have known about it.

**Comment by SteveL-MSFT** [comment]
We have some upcoming docs that should help explain this behavior more clearly, but I don't see anything here that isn't currently by-design
